.drone.star

```diff
@@ -0,0 +1,62 @@
+# To test yaml render before commit:
+#   drone starlark convert --stdout --format
+#
+# To run CI pipeline locally:
+#   drone starlark convert --stdout --format > .drone.yml
+#   drone exec --volume /var/run/docker.sock:/var/run/docker.sock
+
+cmd = 'cargo check'
+deps = 'clang make automake libc-dev libclang-dev pkg-config gnupg protobuf-compiler libgmp-dev nettle-dev'
+
+def main(ctx):
+  return [
+    cargo('format', cmd='cargo fmt -- --check', deps=None, pre=['rustup component add rustfmt']),
```

Comment:
`cargo fmt` fails because it checks module imports, and some protobuf code is generated at build time.

So this will need to be `cargo build && cargo fmt`, which looks like it might be difficult to extend your `cargo` function to do as it runs each command in a separate pipeline?

Reply:
Added it to the `pre` commands arrray. This starlark thing is pretty flexible, you can generate the build pipelines however you like, provided it produces valid dicts at the end

Reply:
Is there a way to generate the protobuf code without building the whole project? Building everything is slow and seems overkill

Reply:
Hmm, could do `cargo check` instead (doesn't produce build output, just checks for errors and warnings). Would still be overkill, but perhaps less so...